op-node/node/node.go

```diff
@@ -189,6 +199,16 @@ func (n *OpNode) initL2(ctx context.Context, cfg *Config, snapshotLog log.Logger
 		return fmt.Errorf("failed to create Engine client: %w", err)
 	}
 
+	// Validate the L2 Client Chain ID
+	if err := cfg.Rollup.CheckL2ChainID(ctx, n.l2Source); err != nil {
+		n.log.Error("failed to verify L2 RPC chain id", "err", err)
+	}
+
```

Comment:
Slight preference to combine these into a single function that checks both the chainid and blockhash, what do you think?